### PR TITLE
Fix a few ESLint errors due to exports oddness

### LINKED ISSUE #1195: Fix ESLint warnings/errors
We switched to ESLint (per #1201), but still have 48 errors which probably needs somebody more familiar with the project to investigate.

```sh
$ npm run lint

> pageshot@0.0.1 lint /Users/pdehaan/dev/github/mozilla-services/pageshot
> eslint .


/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/data/annotate-position.js
  4:10  error  'annotatePosition' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/data/error-utils.js
  15:15  error  Empty block statement  no-empty

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/data/extractor-worker.js
  10:5  error  'FILENAME' is defined but never used         no-unused-vars
  14:7  error  'extractorWorker' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/data/framescripts/add-ids.js
  6:7  error  'addIds' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/data/framescripts/make-static-html.js
  18:7  error  'makeStaticHtml' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/data/framescripts/screenshot.js
   10:7   error  'content' is not defined             no-undef
   11:12  error  'content' is not defined             no-undef
   13:10  error  'content' is not defined             no-undef
   77:1   error  'addMessageListener' is not defined  no-undef
   97:3   error  'sendAsyncMessage' is not defined    no-undef
  100:1   error  'addMessageListener' is not defined  no-undef

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/data/selector-snapping.js
   2:7  error  'snapping' is defined but never used  no-unused-vars
  91:5  error  Unexpected constant condition         no-constant-condition

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/data/selector-ui.js
  6:7  error  'ui' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/data/selector-util.js
  3:7  error  'util' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/lib/helperworker.js
  11:7  error  'captureTab' is defined but never used      no-unused-vars
  15:7  error  'XMLHttpRequest' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/lib/main.js
   33:5  error  'loadReason' is defined but never used  no-unused-vars
  186:5  error  'hotKey' is defined but never used      no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/lib/notificationbox.js
  75:15  error  'on' is defined but never used           no-unused-vars
  75:19  error  'once' is defined but never used         no-unused-vars
  75:25  error  'off' is defined but never used          no-unused-vars
  77:7   error  'timers' is defined but never used       no-unused-vars
  78:26  error  'valid' is defined but never used        no-unused-vars
  84:7   error  'validNumber' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/lib/shooter.js
  26:5  error  'shouldShowTour' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/lib/user.js
  32:5   error  'cachedBackend' is defined but never used  no-unused-vars
  32:20  error  'cachedReason' is defined but never used   no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/chrome-extension/background.js
  13:5  error  'initialized' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/chrome-extension/chrome-shooter.js
  6:7  error  'chromeShooter' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/chrome-extension/randomstring.js
  4:10  error  'randomString' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/chrome-extension/url-domain.js
  6:10  error  'urlDomainForId' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/chrome-extension/uuid.js
  4:10  error  'makeUuid' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/clientglue.js
  132:7  error  'source' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/ga-activation.js
  7:7  error  'React' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/pages/admin/view.js
  2:7  error  'reactruntime' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/pages/shotindex/view.js
  2:7  error  'reactruntime' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/views/export.js
   2:9  error  'Shell' is defined but never used     no-unused-vars
  61:9  error  'TimeDiff' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/views/frame.js
    1:16  error  'Raven' is defined but never used          no-unused-vars
    5:9   error  'ProfileButton' is defined but never used  no-unused-vars
    9:7   error  'ShareButtons' is defined but never used   no-unused-vars
   70:7   error  'Clip' is defined but never used           no-unused-vars
  605:7   error  'ExpireWidget' is defined but never used   no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/views/main.js
  2:9  error  'Shell' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/views/shot-deleted.js
  3:9  error  'Shell' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/static/js/parent-helper.js
  6:3  error  'resolveChildReference' is defined but never used  no-unused-vars

✖ 48 problems (48 errors, 0 warnings)
```

---

#### Original issue:

~~Currently running `$ npm run lint` gives me ~700  errors.
We should tweak the JSHint rules, fix the code, or exclude the offending files.~~

~~A lot of the errors are coming from `static/homepage/js/*`. It looks like we should exclude some/all of those files since they look 3rd party.~~

~~Semi-related, but ESLint is _mucho_ better than JSHint, we may want to consider switching.~~

### PR DESCRIPTION
**Pull request by pdehaan**
#### Current output:

```sh
$ npm run lint

> pageshot@0.0.1 lint /Users/pdehaan/dev/github/mozilla-services/pageshot
> eslint .


/Users/pdehaan/dev/github/mozilla-services/pageshot/addon/lib/main.js
  186:5  error  'hotKey' is defined but never used  no-unused-vars

/Users/pdehaan/dev/github/mozilla-services/pageshot/server/src/views/frame.js
  1:16  error  'Raven' is defined but never used  no-unused-vars

✖ 2 problems (2 errors, 0 warnings)
```

Fixes #1195

### CODE REVIEW

**Comment by pdehaan** [comment]
It'd be nice to also add the following rules to .eslintrc (for increased consistency), but they're a bit more disruptive, so I didn't want to dirty up this PR:

```yaml
  eqeqeq: [1, smart] // "127 warnings"
  quotes: [1, double, {avoidEscape: true}] // "558 warnings"
```

I think ESLint is able to auto-fix the [`quotes`](http://eslint.org/docs/rules/quotes) rule, but I don't think it can auto-fix [`eqeqeq`](http://eslint.org/docs/rules/eqeqeq).

**Comment by pdehaan** [comment]
Good news, Travis-CI is green!
Bad news, not sure if that means anything (ref: #1203).

**Comment by ianb** [comment]
eqeqeq seems like a good idea, but I just can't muster the will to care about which kind of quote marks code uses.

**Comment by fzzzy** [comment]
If it can fix the quotes automatically, then you don't have to muster much will to care. :)

**Comment by pdehaan** [comment]
Yeah, fixing `quotes` is easy and I can do it in about 12 keystrokes.
Fixing the `eqeqeq` can't be done automatically by ESLint, so will require somebody to roll up their sleeves and do it manually (but don't let that discourage you from merging this masterpiece of a PR).

File: addon/data/error-utils.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ function unhandled(error) {
 function getFilename() {
   try {
     return FILENAME;
-  } catch (e) {
+  } catch (e) { // eslint-disable-line no-empty
```
Comment (fzzzy) [inline addon/data/error-utils.js]: Nit: this could also be fixed by moving the return inside the empty block.

**Pull request merged**